src/Illuminate/View/Compilers/Concerns/CompilesComponents.php

```diff
@@ -63,7 +63,7 @@ public static function compileClassComponentOpening(string $component, string $a
     {
         return implode("\n", [
             '<?php if (isset($component)) { $__componentOriginal'.$hash.' = $component; } ?>',
-            '<?php $component = $__env->getContainer()->make('.Str::finish($component, '::class').', '.($data ?: '[]').'); ?>',
+            '<?php $component = $__env->getContainer()->make('.Str::finish($component, '::class').', '.($data ?: '[]').' + (isset($attributes) ? (array) $attributes->getIterator() : [])); ?>',
```

Comment:
Merge component `$data` with (if provided) `$attributes` from parent. The container takes care of the rest.